source/Cosmos.Core_Plugs/System/ActivatorImpl.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using Cosmos.Core;
+using IL2CPU.API.Attribs;
+using IL2CPU.API;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+namespace Cosmos.Core_Plugs.System
+{
+    [Plug(typeof(global::System.Activator))]
+    public static class ActivatorImpl
+    {
+        public static T CreateInstance<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] T>()
+        {
+            return (T)Activator.CreateInstance(typeof(T))!;
+        }
+
+        // nonPublic is ignored since, at the moment, we cannot know if a ctor is public or not
+        // and we would get a Stack Corruption or Null reference exception if the ctor is not present either way.
```

Comment:
is there a way to prevent the Stack Corruption and just throw an exception (we should try and match what .net does whti what exception is thrown)

Reply:
The problem is, we do not have method info, we do not know what method is a ctor and what method is not (and their parameters) at runtime. To make it all work as intended the Vtable should be reworked.

Reply:
What i mean for stack corruption or null reference, is that a method other than the ctor was called or for some reason the null check of the ctor throws (happened while testing but should not happend now). This could happend either because there is no parameterless constructor or that ctor is not at address 0 (till now, it always have been at address 0, but the possibility of it changing is still there).